File: Beecrowd/Beginner./1040.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    double n1, n2, n3, n4, exam, avg, finalAvg;
    
    cin >> n1 >> n2 >> n3 >> n4;

    avg = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10.0;
    
    cout << fixed << setprecision(1);
    cout << "Media: " << avg << endl;

    if (avg >= 7.0) {
        cout << "Aluno aprovado." << endl;
    } else if (avg < 5.0) {
        cout << "Aluno reprovado." << endl;
    } else {
        cout << "Aluno em exame." << endl;
        cin >> exam;
        cout << "Nota do exame: " << exam << endl;
        finalAvg = (avg + exam) / 2.0;
        if (finalAvg >= 5.0) {
            cout << "Aluno aprovado." << endl;
        } else {
            cout << "Aluno reprovado." << endl;
        }
        cout << "Media final: " << finalAvg << endl;
    }

    return 0;
}

File: Beecrowd/1. Beginner/1040/1040.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        double n1, n2, n3, n4, exam, avg, finalavg;

        n1 = input.nextDouble();
        n2 = input.nextDouble();
        n3 = input.nextDouble();
        n4 = input.nextDouble();

        avg = (n1*2 + n2*3 + n3*4 + n4*1) / 10.0;
        avg = Math.floor(avg * 10 + 0.00001) / 10;

        System.out.printf("Media: %.1f\n", avg);

        if (avg >= 7.0) {
            System.out.print("Aluno aprovado.\n");
        } else if (avg < 5.0) {
            System.out.print("Aluno reprovado.\n");
        } else {
            System.out.print("Aluno em exame.\n");
            exam = input.nextDouble();

            System.out.printf("Nota do exame: %.1f\n", exam);
            finalavg = (avg + exam) / 2.0;

            if (finalavg >= 5.0) {
                System.out.println("Aluno aprovado.");
            } else {
                System.out.println("Aluno reprovado.");
            }
            System.out.printf("Media final: %.1f\n", finalavg);
        }
        
        input.close();
    }
}